pkg/daemon/update.go

```diff
@@ -1797,11 +1793,45 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 	return nil
 }
 
+// ExtractAndUpdateOS attempts to consolidating various OS update paths to use the same entrypoint here,
+// to move towards handling CoreOS Layering.
+// Does both extraction (old model) and direct updates (new model)
+// Returns the extracted location to remove
+func ExtractAndUpdateOS(newOSImageURL string) (osImageContentDir string, err error) {
+	if isLayeredUpdate() {
+		// support new flow here, presently going to check for a new field
```

Comment:
Thinking about this a bit more, perhaps instead of inspecting the container dynamically, it makes sense to use a new field in MachineConfig (`osContainer` instead of `osImageURL`?) and in the configmap.

Then we know from quite early on which path we'll take.  We can make everything more "type safe" too...something like:

```
type OSUpdateSource struct {
  old *string
  new *docker.ImageReference
}
```

(Taking the opportunity to use a [proper type for a container image reference](https://pkg.go.dev/github.com/containers/image@v3.0.2+incompatible/docker/reference) instead of `string`)

In Rust of course this would be a nice enum
```
enum OSUpdateSource {
  Old(String)
  New(ImageReference)
}
```
which means the (IMO invalid) states of "no source" and "both sources" are omitted.  But anyways.